app/src/main/java/com/tiptopgoodstudio/androidresources/ui/MainActivity.java

```diff
@@ -2,132 +2,85 @@
 import com.tiptopgoodstudio.androidresources.viewmodel.TopicViewModel;
 
-public class MainActivity extends AppCompatActivity implements SitckyNoteAdapter.ResourceClickListener {
+public class MainActivity extends AppCompatActivity implements StickyNoteAdapter.ResourceClickListener {
 
     private RecyclerView mRecyclerView;
-    private SitckyNoteAdapter mSitckyNoteAdapter;
-    private ProgressBar mLoadingIndicator;
-    private TextView mErrorMessage;
+    private StickyNoteAdapter mStickyNoteAdapter;
+    private View mSplashScreen;
+    private Toolbar mToolbar;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.activity_main);
 
-        Toolbar toolbar = (Toolbar) findViewById(R.id.action_toolbar_main);
-        toolbar.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
-        toolbar.setTitleTextColor(getResources().getColor(R.color.colorMenuItem));
-        setSupportActionBar(toolbar);
+        initializeUiViews();
+        setupActivityAppBar();
+        setupRecyclerView();
+        setupActivityAsDataObserver();
+    }
 
-        mRecyclerView = (RecyclerView) findViewById(R.id.rv_stickynotes);
-        mLoadingIndicator = (ProgressBar) findViewById(R.id.pb_loading_indicator);
-        mErrorMessage = (TextView) findViewById(R.id.main_error_message_display);
+    private void initializeUiViews() {
+        mToolbar = findViewById(R.id.action_toolbar_main);
+        mRecyclerView = findViewById(R.id.rv_stickynotes);
+        mSplashScreen = findViewById(R.id.splash_screen);
+    }
 
+    private void setupRecyclerView() {
         GridLayoutManager layoutManager = new GridLayoutManager(MainActivity.this, 2, GridLayoutManager.VERTICAL, false);
         mRecyclerView.setLayoutManager(layoutManager);
+        mStickyNoteAdapter = new StickyNoteAdapter(this);
+        mRecyclerView.setAdapter(mStickyNoteAdapter);
+    }
 
-        mSitckyNoteAdapter = new SitckyNoteAdapter(this);
-        mRecyclerView.setAdapter(mSitckyNoteAdapter);
-
-        // Load the topics list from the ViewModel
-        loadDataFromViewModel();
+    private void setupActivityAppBar() {
+        mToolbar.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
+        mToolbar.setTitleTextColor(getResources().getColor(R.color.colorMenuItem));
+        setSupportActionBar(mToolbar);
     }
 
     /**
      * This method retrieves data from the ViewModel
-     *
-     * Added by Divya on 4/21/2018
      */
-    private void loadDataFromViewModel(){
-
-        try {
-
-            showProgress();
-
-            TopicViewModel model = ViewModelProviders.of(this).get(TopicViewModel.class);
-
-            model.getTopicsList().observe(this, topicList -> {
-                //Update UI
-                mSitckyNoteAdapter.setTopicData(topicList);
+    private void setupActivityAsDataObserver() {
```

Comment:
The previous code surrounded the call to the ViewModel in a try catch block. Is it alright to not have it?

Reply:
Of all of the examples I have seen of the observer pattern on a ViewModel's LiveData<> instance, I've not seen one in a try-catch block.  Have you seen documentation or code examples which indicate the observer setup needs to go within a try-catch?

Reply:
I have also not seen Observer pattern examples with try catch. However if due to some error, data is not propagated to the UI, is it better to display an error message?

Reply:
With the LiveData pattern a List<Resource> object will absolutely be handed to the adapter through the onChanged object - even if that List happens to have 0 members.  So I don't see an error as being possible under normal user circumstances.  But I could be wrong, so maybe we both go try to find resources to prove me wrong.  I'd much rather know now than once it's pushed out.